src/cli_common.c

```diff
@@ -318,36 +326,26 @@ void parseRedisUri(const char *uri, const char* tool_name, cliConnInfo *connInfo
     UNUSED(tls_flag);
 #endif
 
-    const char *scheme = "valkey://";
-    const char *tlsscheme = "valkeys://";
-    /* We need to support redis:// and rediss:// too for compatibility. */
-    const char *redisScheme = "redis://";
-    const char *redisTlsscheme = "rediss://";
     const char *curr = uri;
     const char *end = uri + strlen(uri);
     const char *userinfo, *username, *port, *host, *path;
 
     /* URI must start with a valid scheme. */
-    if (!strncasecmp(tlsscheme, curr, strlen(tlsscheme))) {
-#ifdef USE_OPENSSL
-        *tls_flag = 1;
-        curr += strlen(tlsscheme);
-#else
-        fprintf(stderr,"valkeys:// and rediss:// are only supported when %s is compiled with OpenSSL\n", tool_name);
-        exit(1);
-#endif
-    } else if (!strncasecmp(redisTlsscheme, curr, strlen(redisTlsscheme))) {
+    if (!strncasecmp(TLS_SCHEME_VALKEY, curr, strlen(TLS_SCHEME_VALKEY)) ||
+        !strncasecmp(TLS_SCHEME_PRE_VALKEY, curr, strlen(TLS_SCHEME_PRE_VALKEY))) {
 #ifdef USE_OPENSSL
         *tls_flag = 1;
-        curr += strlen(redisTlsscheme);
+        char *del = strstr(curr, SCHEME_PREFIX_DELIM);
+        curr += (del - curr) + strlen(SCHEME_PREFIX_DELIM);
```

Comment:
I don't think we need a macro for `"://"`. It unnecessary abstraction that just makes it harder to follow the code. I think we should just use `"://"`. It's shorter and easier to read.